substrate/client/service/src/client/client.rs

```diff
@@ -416,25 +431,16 @@ where
 			backend.commit_operation(op)?;
 		}
 
-		let (unpin_worker_sender, mut rx) =
-			tracing_unbounded::<Block::Hash>("unpin-worker-channel", 10_000);
+		let (unpin_worker_sender, rx) = tracing_unbounded::<UnpinWorkerMessage<Block>>(
+			"notification-pinning-worker-channel",
+			10_000,
+		);
 		let task_backend = Arc::downgrade(&backend);
-		spawn_handle.spawn(
-			"unpin-worker",
-			None,
-			async move {
-				while let Some(message) = rx.next().await {
-					if let Some(backend) = task_backend.upgrade() {
-						backend.unpin_block(message);
-					} else {
-						log::debug!("Terminating unpin-worker, backend reference was dropped.");
-						return
-					}
-				}
-				log::debug!("Terminating unpin-worker, stream terminated.")
-			}
-			.boxed(),
+		let unpin_worker = crate::client::notification_pinning::NotificationPinningWorker::new(
```

Comment:
Nit. maybe you can do the `Arc::downgrade(&backend)` directly in the `NotificationPinningWorker::new` constructor.
Super-Nit : can you import `NotificationPinningWorker` to the scope? :-D